spec/consensus/light-client.md

```diff
@@ -193,33 +193,34 @@ We consider the following set-up:
 we will write ```totalVotingPower(V)``` for ```votingpower_in(V,V)```, which returns the total voting power in V.
 We further use the function ```signers(Commit)``` that returns the set of validators that signed the Commit.
 
-**CheckSupport.** The following function checks whether we can trust the header h2 based on header h1 following the trusting period method.
+**CheckSupport.** The following function checks whether we can trust the header h2 based on header h1 following the trusting period method. Time constraint is
+captured by the `hasExpired` function that depends on trusted period (`tp`) and a parameter `Delta` that denotes minimum duration of header so it is
+not considered expired.
 
 ```go
+  func hasExpired(h) {
+    if h.Header.bfttime + tp - Delta < now { // Observation 1
+      return true
+  }
+
+  // basic validation (function `verify`) has already been called on h2
   func CheckSupport(h1,h2,trustlevel) bool {
-    if h1.Header.bfttime + tp < now { // Observation 1
-      return false // old header was once trusted but it is expired
-    }  
+    if hasExpired(h1) then return false //old header was once trusted but it is expired
+
     vp_all := totalVotingPower(h1.Header.NextV)
-      // total sum of voting power of validators in h2
+    // total sum of voting power of validators in h1
 
     if h2.Header.height == h1.Header.height + 1 {
-      // specific check for adjacent headers; everything must be
-      // properly signed.
-      // also check that h2.Header.V == h1.Header.NextV
-      // Plus the following check that 2/3 of the voting power
-      // in h1 signed h2
-      return (votingpower_in(signers(h2.Commit),h1.Header.NextV) >
-              2/3 * vp_all)
-        // signing validators are more than two third in h1.
-    }
-
-    return (votingpower_in(signers(h2.Commit),h1.Header.NextV) >
-            max(1/3,trustlevel) * vp_all)
-      // get validators in h1 that signed h2
-      // sum of voting powers in h1 of
-      // validators that signed h2
-      // is more than a third in h1
+        // specific check for adjacent headers
+        if h1.Header.NextV == h2.Header.V then
```

Comment:
if this doesn't hold then returning false is probably not ideal since we shouldn't continue with bisection. In my mind CheckSupport() returning false means we should perform further bisection. I believe the checks to verify(), hasExpired() or this check should be done by the caller. Only when these pass should CheckSupport() be called.

Reply:
Alternatively, this function can return a specific error (e.g. `ErrHeaderExpired{h1}`) in such cases